src/react-admin/dataProvider.js

```diff
@@ -1,19 +1,67 @@
+import { fetchUtils } from 'react-admin';
+
+// const apiUrl = process.env.API_URL;
+const apiUrl = process.env.REACT_APP_API_URL + "/admin/";
+const httpClient = fetchUtils.fetchJson;
+
+
+async function deleteOneforEach(currentValue, index, array) {
+  const id = currentValue;
+  const resource = this;
+  const url = `${apiUrl}${resource}/${id}`;
+  const { json } = await httpClient(url, { method: "DELETE" });
+  return { data: json };
+};
+
 export default {
-  getList: (resource, params) => Promise,
+  getList: async (resource, params) => {
+    const url = `${apiUrl}${resource}`;
+    try {
+      const { json, headers} = await httpClient(url);
+      return { data: json, total: json.length };
+    } catch (error) {
+      console.log(error);
+    }
+  },
 
-  getOne: (resource, params) => Promise,
+  getOne: async (resource, params) => {
+    const { id } = params;
+    const url = `${apiUrl}${resource}/${id}`;
+    const { json } = await httpClient(url);
+    return { data: json };
+  },
 
   getMany: (resource, params) => Promise,
 
   getManyReference: (resource, params) => Promise,
 
-  create: (resource, params) => Promise,
+  create: async (resource, params) => {
+    const { data } = params;
+    const url = `${apiUrl}${resource}`;
+    const { json } = await httpClient(url, { method: "POST", body: JSON.stringify(data) });
+    return { data: data, id: 1 };  // Falta el id que tendria que venir en el JSON
+  },
 
-  update: (resource, params) => Promise,
+  update: async (resource, params) => {
+    const {id, data} = params;
+    const url = `${apiUrl}${resource}/${id}`;
+    const { json } = await httpClient(url, { method: "PUT", body: JSON.stringify(data)});
+    return { data: json };
+  },
 
   updateMany: (resource, params) => Promise,
 
-  delete: (resource, params) => Promise,
+  delete: async (resource, params) => {
+    const { id } = params;
+    const url = `${apiUrl}${resource}/${id}`;
+    const { json } = await httpClient(url, { method: "DELETE" });
+    return { data: json };
+  },
 
-  deleteMany: (resource, params) => Promise,
+  deleteMany: async (resource, params) =>  {
+    const { ids } = params;
+    console.log(ids)
+    const data = ids.forEach(deleteOneforEach, resource);
```

Comment:
This will return a pending promise, not the data you are expecting. Take a look at `Promise.all`